src/xunit.performance.api/SafeTerminateHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.Xunit.Performance.Api.Native.Windows;
+using System;
+
+namespace Microsoft.Xunit.Performance.Api
+{
+    internal sealed class SafeTerminateHandler<T> : IDisposable
```

Comment:
I'd recommend calling this something like "SafeDisposableWrapper<T>" or something like that so that it's clear that it's a wrapper around a disposable object, and has semantics similar to "safe" objects like SafeHandle.